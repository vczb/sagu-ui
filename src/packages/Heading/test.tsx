import React from 'react'
import { render, screen } from '../../utils/testUtils'

import Heading from '.'

describe('<Heading/>', () => {
  it('should render the heading', () => {
    render(
      <Heading>
        <span>This is the heading</span>
      </Heading>
    )

    expect(screen.getByText('This is the heading')).toBeInTheDocument()
  })

  it('should render the small heading with lineLeft and lineBottom', () => {
    const { container } = render(
      <Heading size={'small'} lineLeft lineBottom>
        <span>This is the heading</span>
      </Heading>
    )

    expect(container).toMatchSnapshot()
  })

  it('should render the huge heading with lineLeft and lineBottom', () => {
    const { container } = render(
      <Heading size={'huge'} lineLeft lineBottom>
        <span>This is the heading</span>
      </Heading>
    )

    expect(container).toMatchSnapshot()
  })
})
