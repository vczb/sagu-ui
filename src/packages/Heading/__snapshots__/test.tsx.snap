// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should render the huge heading with lineLeft and lineBottom 1`] = `
.c0 {
  color: #222222;
  padding-left: 1.6rem;
  border-left: 0.7rem solid #022B54;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 5.2rem;
}

.c0::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #022B54;
}

<div>
  <h2
    class="c0"
    color="black"
  >
    <span>
      This is the heading
    </span>
  </h2>
</div>
`;

exports[`<Heading/> should render the small heading with lineLeft and lineBottom 1`] = `
.c0 {
  color: #222222;
  padding-left: 1.6rem;
  border-left: 0.7rem solid #022B54;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c0::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #022B54;
}

.c0::after {
  width: 3rem;
}

<div>
  <h2
    class="c0"
    color="black"
  >
    <span>
      This is the heading
    </span>
  </h2>
</div>
`;
